packages/sanity/src/core/util/draftUtils.ts

```diff
@@ -210,3 +210,16 @@ export function removeDupes(documents: SanityDocumentLike[]): SanityDocumentLike
     .map((entry) => entry.draft || entry.published)
     .filter(isNonNullable)
 }
+
+/**
+ * @beta
+ * Given a document id, it indicates whether it is a draft, version or published document.
+ */
+export type DocumentVariantType = 'draft' | 'version' | 'published'
```

Comment:
Speaking of types, I think this probably needs to be moved to operations/types instead of living in the `draftUtils` :)

Reply:
Thanks for the suggestion Rita, I agree with you that `draftUtils` might not be the best place.
However, I think `operation/types `is coupled too much to the `document-pair` and this will be used in more places than that once we have this PR merged https://github.com/sanity-io/sanity/pull/7922

Just pushed a change moving it to its own file inside `core/util/getDocumentVariantType`.

Reply:
Fair enough :)